packages/env/lib/config/parse-config.js

```diff
@@ -97,13 +97,11 @@ function parseSourceString( sourceString, { workDirectoryPath } ) {
 		/^https?:\/\/([^\s$.?#].[^\s]*)\.zip$/
 	);
 
-	if ( zipFields ) {
-		const wpOrgFields = sourceString.match(
-			/^https?:\/\/downloads\.wordpress\.org\/(?:plugin|theme)\/([^\s\.]*)([^\s]*)?\.zip$/
-		);
-		const basename = wpOrgFields
-			? encodeURIComponent( wpOrgFields[ 1 ] )
-			: encodeURIComponent( zipFields[ 1 ] );
+	if ( zipFields && zipFields[ 1 ] ) {
+		// URL: "https://www.example.com/path/to/example-plugin.1.0.2.zip"
+		// Match group: "www.example.com/path/to/example-plugin.1.0.2"
+		// Basename: "example-plugin"
+		const basename = path.basename( zipFields[ 1 ] ).split( '.' )[ 0 ];
```

Comment:
Not sure about removing everything after the first `.`. What if the source string is something like `http://my-server.com/my.cool.plugin.zip`?

Reply:
Fair.. In that case it would resolve to "my". What do you think we should do instead? We could leave periods in the filename (will this be supported everywhere?), or perhaps convert them to dashes?

Reply:
Periods are generally allowed, no?

Maybe remove the numbers from the end if it's a w.org URL and leave everything as is if not.

- `https://example.com/my.cool.plugin.4.6.0.zip` → `my.cool.plugin.4.6.0`
- `https://downloads.wordpress.org/plugin/my.cool.plugin.4.6.0.zip` → `my.cool.plugin`